test/integration/retryable-reads/retryable_reads.spec.prose.test.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { expect } from 'chai';
+
+import { Collection, MongoClient } from '../../../src';
+
+describe('Retryable Reads Spec Prose', () => {
+  let client: MongoClient, failPointName;
+
+  afterEach(async () => {
+    try {
+      if (failPointName) {
+        await client.db('admin').command({ configureFailPoint: failPointName, mode: 'off' });
+      }
+    } finally {
+      failPointName = undefined;
+      await client?.close();
+    }
+  });
+
+  describe('PoolClearedError Retryability Test', () => {
+    // This test will be used to ensure drivers properly retry after encountering PoolClearedErrors.
+    // It MUST be implemented by any driver that implements the CMAP specification.
+    // This test requires MongoDB 4.2.9+ for blockConnection support in the failpoint.
+
+    let cmapEvents: Array<{ name: string; event: Record<string, any> }>;
+    let commandStartedEvents: Array<Record<string, any>>;
+    let testCollection: Collection;
+    beforeEach(async function () {
+      // 1. Create a client with maxPoolSize=1 and retryReads=true.
+      // If testing against a sharded deployment, be sure to connect to only a single mongos. <-- TODO: what does that look like?
```

Comment:
Optional, but you can use the uri helper passed to newClient:
```ts
client = this.configuration.newClient(this.configuration.uri({
        maxPoolSize: 1,
        retryReads: true,
        monitorCommands: true,
        useMultipleMongoses: false // If testing against a sharded deployment, be sure to connect to only a single mongos.
}))
```

if you want to lock in the single mongos behavior rather than depending on the default